packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmObjectHelper.kt

```diff
@@ -236,26 +258,33 @@ internal object RealmObjectHelper {
         } catch (exception: Throwable) {
             throw CoreExceptionConverter.convertToPublicException(exception) { coreException: RealmCoreException ->
                 when (coreException) {
-                    is RealmCorePropertyNotNullableException -> {
+                    is RealmCorePropertyNotNullableException ->
                         IllegalArgumentException("Required property `${obj.className}.${obj.metadata[key]!!.name}` cannot be null")
-                    }
-                    is RealmCorePropertyTypeMismatchException -> {
+                    is RealmCorePropertyTypeMismatchException ->
                         IllegalArgumentException("Property `${obj.className}.${obj.metadata[key]!!.name}` cannot be assigned with value '${value.value}' of wrong type")
-                    }
-                    is RealmCoreLogicException -> {
-                        IllegalArgumentException("Property `${obj.className}.${obj.metadata[key]!!.name}` cannot be assigned with value '${value.value}': ${coreException.message}")
-                    }
-                    else -> {
-                        throw IllegalStateException(
-                            "Cannot set `${obj.className}.$${obj.metadata[key]!!.name}` to `${value.value}`: changing Realm data can only be done on a live object from inside a write transaction. Frozen objects can be turned into live using the 'MutableRealm.findLatest(obj)' API.",
-                            exception
-                        )
-                    }
+                    is RealmCoreLogicException -> IllegalArgumentException(
+                        "Property `${obj.className}.${obj.metadata[key]!!.name}` cannot be assigned with value '${value.value}'",
+                        exception
+                    )
+                    else -> IllegalStateException(
+                        "Cannot set `${obj.className}.$${obj.metadata[key]!!.name}` to `${value.value}`: $NOT_IN_A_TRANSACTION_MSG",
+                        exception
+                    )
                 }
             }
         }
     }
 
+    internal fun setMutableInt(
```

Comment:
Do we really need specialized versions of this?! I would imagine that we should just define an appropriate converter. Maybe it isn't obvious and worth it since `MutableRealmInt` isn't supported in the collection types, but I would prefer to follow the generic pattern until we decide to do specialized paths for any of the types.

Reply:
I've tried and it's possible to return unmanaged instances using the current framework (whose behaviour I wasn't very familiar with), though getting it to work for managed instances requires some non-trivial changes in the compiler plugin. I will try to devise a solution that doesn't turn into a timesink, otherwise I will leave it as is and will add a todo and create a task for it.